### ISSUE TITLE
run_tests.py fails with an OSError

### ISSUE THREAD

**Issue by murgatroid99** [grpc/grpc#3930]
Seen here: https://grpc-testing.appspot.com/job/gRPC_pull_requests/3929/config=dbg,language=php,platform=macos/console

```
START: tools/run_tests/build_php.sh
Traceback (most recent call last):
  File "./tools/run_tests/run_tests.py", line 893, in <module>
    xml_report=args.xml_report)
  File "./tools/run_tests/run_tests.py", line 807, in _build_and_run
    newline_on_success=newline_on_success, travis=travis)
  File "/jenkins/workspace/gRPC_pull_requests/config/dbg/language/php/platform/macos/tools/run_tests/jobset.py", line 428, in run
    if not js.start(cmdline):
  File "/jenkins/workspace/gRPC_pull_requests/config/dbg/language/php/platform/macos/tools/run_tests/jobset.py", line 346, in start
    self._xml_report)
  File "/jenkins/workspace/gRPC_pull_requests/config/dbg/language/php/platform/macos/tools/run_tests/jobset.py", line 205, in __init__
    self.start()
  File "/jenkins/workspace/gRPC_pull_requests/config/dbg/language/php/platform/macos/tools/run_tests/jobset.py", line 221, in start
    env=env)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 1316, in _execute_child
    data = _eintr_retry_call(os.read, errpipe_read, 1048576)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 476, in _eintr_retry_call
    return func(*args)
OSError: [Errno 22] Invalid argument
```

**Comment by jtattermusch** [comment]
Do we still keep seeing this?

**Comment by adelez** [comment]
I think it was fixed by #4046
